powershell/risk_meters/ShowHistoricalVulnCounts.ps1

```diff
@@ -0,0 +1,286 @@
+        [String]$startDate
+    )
+    
+    $Resp = Invoke-Risk-Meter-Report -BaseUrl $BaseUrl -Headers $Headers -Id $Id `
+                                     -Report "historical_open_vulnerability_count_by_risk_level" -startDate $startDate
+
+    Return $Resp
+}
+
+function Invoke-Historical-Past-Due-Vuln-Counts
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [String]$BaseUrl,
+
+        [Parameter(Mandatory)]
+        [Object]$Headers,
+
+        [Parameter(Mandatory)]
+        [Int64]$Id,
+
+        [Parameter(Mandatory)]
+        [String]$startDate
+    )
+    
+    $Resp = Invoke-Risk-Meter-Report -BaseUrl $BaseUrl -Headers $Headers -Id $Id `
+                                     -Report "historical_past_due_vulnerabilities_by_risk_level" -startDate $startDate
+
+    Return $Resp
+}
+# main
+Write-Host "Historical Risk Meter Vulnerability Counts"
+Write-Host ""
+
+# Obtain the Kenna Security API key from an environment variable.
+$KennaApiKey = ""
+Try {
+    $KennaApiKey = (Get-ChildItem -Path Env:\KENNA_API_KEY -ErrorAction Stop).Value
+}
+Catch {
+    Write-Host "API key is non-existent"
+    Exit
+}
+
+# HTTP headers.
+$Headers = @{
+    "Accept" = "application/json"
+    "Content-Type" = "application/json; charset=utf-8"
+    "X-Risk-Token" = $KennaApiKey
+}
+
+# You might have to change this depending on your deployment.
+$BaseUrl = "https://api.kennasecurity.com/"
+
+# Obtain a list of risk meters.
+$AssetGroups = Invoke-List-Risk-Meters -BaseUrl $BaseUrl -Headers $Headers
+
+#$AssetGroups | Format-Table -Property name, id, asset_count, @{Label="Score"; Expression={$_.risk_meter_score}}, updated_at
+
+# For each asset group, obtain the total number of vulnerabilities (vulns) and the number of vulns past the due date.
+ForEach ($AssetGroup in $AssetGroups) {
+    # Invoke APIs to obtain the historical vuln counts.
+    $VulnCountsResp = Invoke-Historical-Vuln-Counts -BaseUrl $BaseUrl -Headers $Headers -Id $AssetGroup.id -startDate $startDate
+    $PastDueVulnCountsResp = Invoke-Historical-Past-Due-Vuln-Counts -BaseUrl $BaseUrl -Headers $Headers `
+                                                                    -Id $AssetGroup.id -startDate $startDate
+
+    #Write-Host "$($AssetGroup.name)  $($AssetGroup.id)  $($AssetGroup.asset_count)  $($AssetGroup.risk_meter_score)"
+    # Verify that the Asset Group IDs match.
+    If ($VulnCountsResp.id -ne $PastDueVulnCountsResp.id) {
+        Write-Howt "IDs don't match" -ForegroundColor Red
+        Exit
+    }
+
+    # Acquire the vuln counts by dates.
+    $HistoricalVulnCounts = $VulnCountsResp.historical_vulnerability_count_by_risk
+    $HistoricalPastDueVulnCounts = $PastDueVulnCountsResp.historical_past_due_vulnerabilities_by_risk_level
+
+    # Keep only the first day of the months for total vuln counts.
+    $keyHash = [ordered] @{}
+    $HistoricalVulnCounts.PsObject.Properties | ForEach-Object {
+        if ($_.Name -match '\d\d\d\d-\d\d-01') {

```

Comment:
Just FYI, it matches probably without beginning and end of line. Also, there is `\d{4}` syntax, for example

```
$ irb
> "asdasd1231-12-01" =~ /\d{4}-\d{2}-01/
 => 6
 ```

Reply:
Maybe something like `/\d{4}-\d{1,2}-01/` is even better, so it can catch 2022-2-01

Reply:
Good suggestion, but it is not the case.